### PR TITLE
feat(search): included solr endpoint support for search

### PR DESCRIPTION
**Pull request by riginoommen**
## Resolves
ONEPLAT714

### 

## Explain the feature
Added SOLR endpoint to search microservice and updated the data structure to the SPA and microservices.

## Does this PR introduce a breaking change

No

## Screenshots

NA

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

**Review by sayak-sarkar — CHANGES_REQUESTED** [review]
@riginoommen Please go through the LGTM alerts and see if there is a better way of tackling these certificate issues.

**Comment by riginoommen** [comment]
> @riginoommen Please go through the LGTM alerts and see if there is a better way of tackling these certificate issues.

HI Sayak,

Removed the reject unauthorozed code and made the exceptions in the environment variables

File: packages/search-service/.env.example [hunk 1]
```diff
@@ -19,3 +19,6 @@ COOKIE_PARSER=SEARCH_SECRET
 
 ## Authorization Token
 # AUTH_TOKEN=
+
+
+# NODE_TLS_REJECT_UNAUTHORIZED
```
Comment (sayak-sarkar) [inline packages/search-service/.env.example:24]:
```suggestion
# NODE_TLS_REJECT_UNAUTHORIZED

```

**Review by sayak-sarkar — CHANGES_REQUESTED** [review]

**Review by ghanlohar — APPROVED** [review]
Looks good to me.

**Review by sayak-sarkar — APPROVED** [review]

**Comment by sayak-sarkar** [comment]
@riginoommen Please fix the PR's title as per the team norms doc!

**Review by sayak-sarkar — APPROVED** [review]

**Pull request merged**